__tests__/Ship.test.js

```diff
@@ -0,0 +1,86 @@
+const { Ship } = require("../src/Ship")
+
+describe("Ship", () => {
+  describe("with ports and an itinerary", () => {
+    let firstPort, secondPort, lastPort, itinerary, ship
+
+    beforeEach(() => {
+      firstPort = { addShip: jest.fn(), removeShip: jest.fn() }
```

Comment:
Once more using a mock function which does not represent what the real use case is, this should be updated